govcd/vm_test.go

```diff
@@ -494,11 +494,22 @@ func (vcd *TestVCD) Test_HandleInsertOrEjectMedia(check *C) {
 	check.Assert(err, IsNil)
 
 	//verify
-	err = vm.Refresh()
-	check.Assert(err, IsNil)
+	vcd.waitUntilEjected(err, vm, check)
 	check.Assert(isMediaInjected(vm.VM.VirtualHardwareSection.Item), Equals, false)
 }
 
+// waitUntilEjected checks status until ready. We need this cause VCD is slow to change representative state
+func (vcd *TestVCD) waitUntilEjected(err error, vm *VM, check *C) {
```

Comment:
While this function makes the test pass, it doesn't solve the problem.
The problem is that with some recent vCD versions, the refresh doesn't return quickly enough.
Tests should certify that the SDK woks as expected. If they don't, we should fix the root cause, not change the test to adapt it to the symptoms/

Reply:
Well, issue what refresh return to quickly enough.
In general I think it's VCD bug and we should report it.
Workaround in SDK would be possible if we in detach function would fetch refresh untill VCD database updated.